challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |string| string.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all { |i|  i.downcase == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.select { |i|  i.downcase != "dog"  }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count  
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.reject { |i| i if !i.downcase.start_with? 'c' and !i.downcase.start_with? 'd' }
```

Comment:
`reject` along with negation inside the block is a double negative. Phrasing it positively makes it easier for the next developer to read.

[`String#start_with?`](http://ruby-doc.org/core-2.4.2/String.html#method-i-start_with-3F) can check multiple prefixes if you provide multiple arguments.